__tests__/pages/exercises/[lessonSlug].test.js

```diff
@@ -1,12 +1,122 @@
 import React from 'react'
-import { render, waitFor } from '@testing-library/react'
+import {
+  render,
+  waitFor,
+  screen,
+  waitForElementToBeRemoved
+} from '@testing-library/react'
 import '@testing-library/jest-dom'
 import Exercises from '../../../pages/exercises/[lessonSlug]'
+import { useRouter } from 'next/router'
+import { MockedProvider } from '@apollo/client/testing'
+import dummyLessonData from '../../../__dummy__/lessonData'
+import dummySessionData from '../../../__dummy__/sessionData'
+import dummyAlertData from '../../../__dummy__/alertData'
+import GET_APP from '../../../graphql/queries/getApp'
+
+const session = {
+  ...dummySessionData,
+  submissions: [],
+  lessonStatus: []
+}
 
 describe('Exercises page', () => {
+  const { query } = useRouter()
+  query['lessonSlug'] = 'js0'
   test('Should render correctly', async () => {
-    const { getByRole } = render(<Exercises />)
+    const mocks = [
+      {
+        request: { query: GET_APP },
+        result: {
+          data: {
+            session,
+            lessons: dummyLessonData,
+            alerts: dummyAlertData
+          }
+        }
+      }
+    ]
+
+    const { getByRole } = render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Exercises />
+      </MockedProvider>
+    )
+
+    await waitFor(() =>
+      getByRole('heading', { name: /Foundations of JavaScript/i })
+    )
+  })
+
+  test('Should render a 500 error page if the lesson data is null', async () => {
+    const mocks = [
+      {
+        request: { query: GET_APP },
+        result: {
+          data: {
+            session,
+            lessons: null,
+            alerts: dummyAlertData
+          }
+        }
+      }
+    ]
+
+    const { getByRole } = render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Exercises />
+      </MockedProvider>
+    )
+
+    await waitFor(() => getByRole('heading', { name: /500 Error/i }))
```

Comment:
This is not wrapped in `expect` because `getByRole` will throw an error if the element isn't found?

Reply:
Yes, getByRole throws an error if the element isn't found, however queryByRole doesn't throw if the element isn't found.